### PR TITLE
tensorflow-io v0.8.0 release

### LINKED ISSUE #459: tensorflow-io 0.8.0 release
This is to start the planning for the release of 0.8.0 for tensorflow-io.

Here is the list of items for 0.8.0:

- [x] Fix test errors in #431 
- [x] Update the version in setup.py
- [x] Update README.md to add additional supported ops since 0.7.0
- [x] Update RELEASE.md to capture 0.8.0 changes and version compatibility.
- [x] Push *.whl files to PyPI.org, including macOS package.
- [x] Release R package to CRAN (optionally).

/cc @yongtang @terrytangyuan @dmitrievanthony

### PR DESCRIPTION
**Pull request by yongtang**
NOTE: This PR should not be merged until v0.8.0 is released.

This PR is from R0.8 branch and need to do the following:

1) Wait until tensorflow==1.15.0 is released and update the version in setup.py.
2) Rebased against master if necessary
3) Make sure all tests pass
4) Complete all outstanding items listed in #459 (comment)
5) Release v0.8.0 with tag created from R0.8 branch
6) Merge this PR into master branch.
7) Delete R0.8 branch.

This PR fixes #459

/cc @BryanCutler @terrytangyuan @dmitrievanthony

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Comment by yongtang** [comment]
Some of the tests failures are caused by 1.15.0 and 2.0.0 discrepancy which is being addressed in PR #495.

**Comment by yongtang** [comment]
Rebased with #495 merged. Tests should pass now.

**Comment by yongtang** [comment]
Still have one more test failure. Will fix shortly.

**Comment by yongtang** [comment]
All tests pass now.

**Comment by yongtang** [comment]
RELEASE.md has been updated with the following:
```
# Release 0.8.0

## Major Features
* `tensorflow_io.json`: JSON Dataset support.
* `tensorflow_io.dicom`: Medical image DICOM format support.
* `tensorflow_io.genome`: DNA Sequence FastQ format support added.
* `tensorflow_io.ffmpeg`: FFmpeg now support selection of video, audio,
  and subtitle sub streams.
* `tensorflow_io.ffmpeg`: FFmpeg now support decode video from memory.
* `tensorflow_io.kafka`: Kafka Dataset now support Kafka Schema Registry.
* `tensorflow_io.audio`: WAV format now support 24 bit audio streams.
* `tensorflow_io.text`: Regex capture group (`re2_full_match`) support.
* manylinux2010 compliant on Linux.

## Thanks to our Contributors

This release contains contributions from many people:

Aleksey Vlasenko, Bryan Cutler, Damien Pontifex, Duke Wang, Jiacheng Xu,
Marcelo Lerendegui, Mark Daoust, Ouwen Huang, Suyash Kumar, Yong Tang,
Yuan Tang, android

We are also grateful to all who filed issues or helped resolve them, asked and
answered questions, and were part of inspiring discussions.

```

Let me know if I missed any items.

**Comment by yongtang** [comment]
We also need to rebuild the Travis CI.

**Comment by yongtang** [comment]
TensorFlow 1.15.0 has been released. As such this PR has been updated (bumped to `tensorflow>=1.15.0,<1.16.0`) and is ready for review.

/cc @terrytangyuan @BryanCutler @vlasenkoalexey @dmitrievanthony

File: RELEASE.md [hunk 1]
```diff
@@ -1,3 +1,29 @@
...
+* `tensorflow_io.text`: Regex capture group (`re2_full_match`) support.
+* manylinux2010 compliant on Linux.
+
+## Thanks to our Contributors
+
+This release contains contributions from many people:
+
+Aleksey Vlasenko, Amarpreet Singh, Bryan Cutler, Damien Pontifex, Duke Wang,
+Jiacheng Xu, Marcelo Lerendegui, Mark Daoust, Ouwen Huang, Suyash Kumar,
+Yong Tang, Yuan Tang, android
```
Comment (terrytangyuan) [inline RELEASE.md]: Who's "android"?

Comment (yongtang) [reply RELEASE.md]: @terrytangyuan android (henrytansetiawan).

Comment (terrytangyuan) [reply RELEASE.md]: We should probably used henrytansetiawan instead? I understand that this list is generated automatically but we should probably manually edit it if needed.

Comment (yongtang) [reply RELEASE.md]: @terrytangyuan Update.

**Comment by yongtang** [comment]
Looks like 1.15.0 revert back the namespace change and `IO>` is not allowed anymore (was working with 11.5.0rc3) :cry:

As a workaround, I will change `IO>` to `Io` so that no python code change needed. Once we release 0.8.0 I will revert back from `Io` to `IO>`.

**Comment by yongtang** [comment]
All tests pass now 🎉 , I will merge this PR, then have a commit to trigger Travis CI cache. The Travis CI cache is becoming a bottleneck, thought this is what we could do at the moment until we have the Bazel remote cache in place (as was discussed in our last meeting).

**Pull request merged**